tm2py/components/network/highway/highway_assign.py

```diff
@@ -140,65 +140,77 @@ def run(self):
-                vdfs = [
-                    f for f in scenario.emmebank.functions() if f.type == "VOLUME_DELAY"
-                ]
-                for function in vdfs:
-                    expression = function.expression
-                    for el in ["el1", "el2", "el3", "el4"]:
-                        expression = expression.replace(el, getattr(exf_pars, el))
-                    if "@static_rel" in expression:
-                        # split function into time component and reliability component
-                        time_expr, reliability_expr = expression.split(
-                            "*(1+@static_rel+"
+                    with self.logger.log_start_end(
+                        "Run SOLA assignment without path analyses", level="INFO"
+                    ):
+                        assign = self.controller.emme_manager.tool(
+                            "inro.emme.traffic_assignment.sola_traffic_assignment"
                         )
-                        net_calc(
-                            "@auto_time",
-                            time_expr,
-                            {"link": "vdf=%s" % function.id[2:]},
-                        )
-                        net_calc(
-                            "@reliability",
-                            "(@static_rel+" + reliability_expr,
-                            {"link": "vdf=%s" % function.id[2:]},
-                        )
-                        net_calc("@reliability_sq", "@reliability**2", {"link": "all"})
+                        assign(assign_spec, scenario, chart_log_interval=1)
+
+                    # calucaltes link level LOS based reliability
+                    net_calc = NetworkCalculator(self.controller, scenario)
+
+                    exf_pars = scenario.emmebank.extra_function_parameters
+                    vdfs = [
+                        f
+                        for f in scenario.emmebank.functions()
+                        if f.type == "VOLUME_DELAY"
+                    ]
+                    for function in vdfs:
+                        expression = function.expression
+                        for el in ["el1", "el2", "el3", "el4"]:
+                            expression = expression.replace(el, getattr(exf_pars, el))
+                        if "@static_rel" in expression:
+                            # split function into time component and reliability component
+                            time_expr, reliability_expr = expression.split(
+                                "*(1+@static_rel+"
+                            )
+                            net_calc(
+                                "@auto_time",
+                                time_expr,
+                                {"link": "vdf=%s" % function.id[2:]},
+                            )
+                            net_calc(
+                                "@reliability",
+                                "(@static_rel+" + reliability_expr,
```

Comment:
@inrokevin I noticed `@reliability == @static_rel` for all links in the final assigned network which means the LOS based part of reliability was 0 in this step. I believe it's because here `reliability_expr` had get(1), but get(1) was not initialized. I removed the get(1) from the reliability expr. Please give a thumb-up if you agree with this approach.

Reply:
To clarify, this is after I initialized put(1) in the BPR and Akcelik parts in the VDFs.

Reply:
Agree, this is the simplest fix. I reviewed the changed expression and it looks correct. Thanks Sijia.